### PR TITLE
SSL support for HTTPS and WSS

### PR DESCRIPTION
**Pull request by giuliojiang**
Can launch servers in secure mode
```
./RunServer start https
```
Use Firefox to accept the self-signed certificates by visiting
https://localhost:4444
and then
https://localhost:4443

### CODE REVIEW

**Comment by giuliojiang** [comment]
this was mainly for the WebSocket server. Getting a HTTPS server working is easy.
I'm going to switch to Flask webserver anyways, and this was done because if the webserver is HTTPS, then the WebSocket server must also use SSL.
We want to have our servers ready to be integrated in a real secure environment.

File: frontend/webclient/script/socket.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 var $globals = {};
 
+var ws_address = location.protocol == 'https:' ?
```
Comment (ke00n) [inline frontend/webclient/script/socket.js]: :top:

File: infrastructure/infrastructure/src/main/java/alabno/wserver/Main.java [hunk 2]
```diff
@@ -1,27 +1,65 @@
...
+
+        System.out.println("Starting WebSocket server on port " + port);
+        AutoMarkerWSServer the_server = new AutoMarkerWSServer(port);
+
+        if (secure) {
+            // Set up the WebSocket server in secure mode
+            String STORETYPE = "JKS";
+            String KEYSTORE = "frontend/mykeystore.jks";
+            String STOREPASSWORD = "albano";
+            String KEYPASSWORD = "albano";
```
Comment (ke00n) [inline infrastructure/infrastructure/src/main/java/alabno/wserver/Main.java]: probably a dumb remark, but there two values (STOREPASSWORD and KEYPASSWORD) are tmp right?

Comment (giuliojiang) [inline infrastructure/infrastructure/src/main/java/alabno/wserver/Main.java]: yes

File: frontend/httpserver.py [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+import ssl
+import os
+import sys
 
-PORT = 8000
+if len(sys.argv) > 1 and sys.argv[1] == 'https':
+    exec_dir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)))
 
-Handler = SimpleHTTPServer.SimpleHTTPRequestHandler
+    httpd = BaseHTTPServer.HTTPServer(('', 4443), SimpleHTTPServer.SimpleHTTPRequestHandler)
```
Comment (AbraaoMota) [inline frontend/httpserver.py]: https?

Comment (helicopter88) [inline frontend/httpserver.py]: <3

**Pull request merged**